src/main/java/org/testng/TestRunner.java

```diff
@@ -1225,6 +1226,24 @@ public Object removeAttribute(String name) {
     return m_guiceModules.get(cls);
   }
 
+  @Override
+  public List<Module> getGuiceModules() {
+    return m_guiceModules.values().
+        stream()
+        .flatMap(Collection::stream)
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public void addGuiceModule(Module module) {
+    Class<? extends Module> cls = module.getClass();
+    List<Module> modules = m_guiceModules.get(cls);
+    boolean found = modules.stream().anyMatch(each -> each.getClass().equals(cls));
```

Comment:
I feel there is an issue if a module is found because it is not supposed to be added.
Does it still work if an exception is thrown when found?

Reply:
I didnt quite understand this. I am basically trying to implement "Add module if it already doesn't exist". This comes up only when we end up instantiating the module multiple times.

Reply:
In fact, I don't understand why we would like to add an already existing module. I think it should be avoided if possible.

Reply:
@juherr - There are multiple places from wherein `addModule()` is being attempted and in many of them, the knowledge of "add only if not exists" does not reside. So adding that edit check in this one place ensures that we dont slip in duplicates.

Reply:
Ok, I understand. I'm not feeling it good because it is hiding potential issues.
BTW, it is not a blocking issue for me.